File: UserProfileService/UserProfileService.Application/DTOs/DishDTOs/CreateDishDTO.cs
using Microsoft.AspNetCore.Http;

namespace UserProfileService.Application.DTOs
{
    public class CreateDishDTO
    {
        public int ProfileId { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public int AmountOfPortions { get; set; }

        public List<IngredientOfDishDTO> Ingredients { get; set; } = new List<IngredientOfDishDTO>();
    }
}

File: NutritionBalance.Client/src/app/models/profile/DTOs/dish/create-dish.dto.ts
import { IngredientOfDishDTO } from './ingredient-of-dish.dto';

export class CreateDishDTO {
  profileId!: number;
  name!: string;
  description?: string;
  amountOfPortions!: number;
  ingredients: IngredientOfDishDTO[] = [];
}
